packages/web3-providers/lib/providers/AbstractSocketProvider.js

```diff
@@ -165,8 +165,6 @@ export default class AbstractSocketProvider extends EventEmitter {
     onError(error) {
         this.emit(this.ERROR, error);
         this.emit(this.SOCKET_ERROR, error);
-        this.removeAllSocketListeners();
```

Comment:
On reconnection new socket created each time so we need to clear socket events

Reply:
I'm currently testing the memory consumption etc. and will do the finish. Could you please wait until the ``in-progress`` got removed. Thanks.